### PR TITLE
Fix MTEB meta and add test for retrieval tasks

### LINKED ISSUE #885: Use the model similarity instead of benchmark defined similarity
We currently use a similarity metric decided by the benchmark instead of by the model (also discussed in #766). Sentence transformers v3 allow for using a custom similarity metric.

### LINKED ISSUE #903: Changes in metadata broke backward compat of metric name
The metric name for STS & co changed in the metadata as previously it was created via merging the `cos_sim` & `spearman` names i.e. https://github.com/embeddings-benchmark/mteb/blob/2cc222efd84c29cbb5ff04fb8e6703674d53dda9/scripts/mteb_meta.py#L113 and now it is defined as `cosine_spearman` i.e. https://github.com/embeddings-benchmark/mteb/blob/b16d7d5620b987992e16ee663e5d82ee007acd8e/mteb/tasks/STS/eng/BiossesSTS.py#L21C21-L21C36 which is then taken as is i.e. https://github.com/embeddings-benchmark/mteb/blob/b16d7d5620b987992e16ee663e5d82ee007acd8e/mteb/cli.py#L265

For old models like https://huggingface.co/w601sxs/b1ade-embed it was called `cos_sim_spearman`.
For new models like https://huggingface.co/w601sxs/b1ade-embed-kd_3 it is now called `cosine_spearman`.

This now breaks the leaderboard as people are starting to submit with the new metric name but the old one is asked for in the code via https://huggingface.co/spaces/mteb/leaderboard/blob/a812c3bbcc3cd88510b60c6c5788d95668d69dc6/config.yaml#L38 which is then used to get the scores via 
 https://huggingface.co/spaces/mteb/leaderboard/blob/a812c3bbcc3cd88510b60c6c5788d95668d69dc6/app.py#L245

Same will happen for PairClassification and others I think. 🤔

Options are:
1) Allow both metric names; Requires some changes to the leaderboard code
2) Revert `cosine_spearman` to `cos_sim_spearman` & others
3) Maybe sth else reasonable that I'm missing?

Not sure what's best, cc @KennethEnevoldsen & @tomaarsen who probably know better! 🙌

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
This pull request fixes the MTEB meta to include all scores and adds a test for retrieval and STS tasks using the old results format.

It also:
- adds additional normalization to historic results (notably for STS).
- normalizes existing STS results to a flat format be in line with main_score
- add custom similarity functions to models 

Fixes #903 
Partly fixes #885 

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

**Review by Muennighoff — COMMENTED** [review]
Amazing work on this! 🚀 Main thing is reg. the language / hf subset codes 🤔

File: tests/create_meta/model_card_gold.md [hunk 1]
```diff
@@ -2,17 +2,510 @@
...
+      value: 0.13082283087945362
+    - type: euclidean_spearman
+      value: 0.2263866797712348
+    - type: main_score
+      value: 0.2263866797712348
+    task:
+      type: STS
+  - dataset:
+      config: de
+      name: MTEB STS22
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply tests/create_meta/model_card_gold.md:120]: (1) seems like the easiest to implement. Languages would be the most human-interpretable (but as you say, some may contain multiple, and how to handle these is not quite clear), and often, hf_subsets are also reasonable).

File: mteb/models/__init__.py [hunk 2]
```diff
@@ -91,6 +91,7 @@ def model_meta_from_sentence_transformers(model: SentenceTransformer) -> ModelMe
             release_date=None,
             languages=languages,
             framework=["Sentence Transformers"],
+            distance_metric=model.similarity_fn_name,
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/models/__init__.py:94]: I am all for compatibility - changing it

File: mteb/evaluation/evaluators/STSEvaluator.py [hunk 3]
```diff
@@ -65,17 +68,33 @@ def __call__(self, model):
         euclidean_pearson, _ = pearsonr(self.gold_scores, euclidean_distances)
         euclidean_spearman, _ = spearmanr(self.gold_scores, euclidean_distances)
 
+        similarity_scores = None
+        if hasattr(model, "similarity_pairwise"):
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/evaluation/evaluators/STSEvaluator.py:72]: Ahh yes - I have added these as well

File: tests/create_meta/model_card_gold.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply tests/create_meta/model_card_gold.md:120]: (added hf subset score for now). This adds it in all cases even when it is default.

**Pull request merged**